src/vulnpy/django/vulnerable.py

```diff
@@ -6,78 +8,79 @@
     from django.conf.urls import url as compat_url
 
 from vulnpy.common import get_template
-from vulnpy.trigger import cmdi, deserialization
+from vulnpy.trigger import TRIGGER_MAP, get_trigger, cmdi, deserialization  # noqa: F401
 
 
-def _vulnpy_root(request):
-    return HttpResponse(get_template("home.html"))
+def _get_user_input(request):
+    if request.method == "GET":
+        return request.GET.get("user_input", "")
+    return request.POST.get("user_input", "")
 
 
-def _cmdi(request):
-    return HttpResponse(get_template("cmdi.html"))
+def gen_root_view(name):
+    def _root(request):
+        return HttpResponse(get_template("{}.html".format(name)))
 
+    return _root
 
-def _deserialization(request):
-    return HttpResponse(get_template("deserialization.html"))
 
+def get_trigger_view(name, trigger):
+    def _view(request):
+        user_input = _get_user_input(request)
 
-def _cmdi_os_system(request):
-    user_input = _get_user_input(request)
-    cmdi.do_os_system(user_input)
-    return HttpResponse(get_template("cmdi.html"))
+        module = sys.modules.get("vulnpy.trigger.{}".format(name))
+        if not module:
+            raise RuntimeError(
+                "Please import trigger module {} at the top of {}".format(
+                    name, __file__
+                )
+            )
 
+        trigger_func = get_trigger(module, trigger)
 
-def _cmdi_subprocess_popen(request):
-    user_input = _get_user_input(request)
-    cmdi.do_subprocess_popen(user_input)
-    return HttpResponse(get_template("cmdi.html"))
+        if trigger_func:
+            trigger_func(user_input)
 
+        return HttpResponse(get_template("{}.html".format(name)))
 
-def _deserialization_pickle_load(request):
-    user_input = _get_user_input(request)
-    deserialization.do_pickle_load(user_input)
-    return HttpResponse(get_template("deserialization.html"))
+    return _view
 
 
-def _deserialization_pickle_loads(request):
-    user_input = _get_user_input(request)
-    deserialization.do_pickle_loads(user_input)
-    return HttpResponse(get_template("deserialization.html"))
+def get_root_name(name):
+    if name == "home":
```

Comment:
I'm noticing a few special cases for `"home"`, and I think it might make more sense for this route not to be autogenerated. That way we know when we're in code that's generating routes we won't keep running into this special case

Reply:
I thought about this over the weekend and I actually think there is another good reason not to automate the generation of the Home view - in case we want to add more stuff to home!

Reply:
Ok so I think I"ve changed my mind. I think it actually is more reasonable to keep the implementation as is with the if name == "home" checks for consistency. We can iterate on this later on if needed, but I don't think the home view is important enough to have its own pattern.

Reply:
you're more familiar with this than I am so I trust your judgement